flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/utils/ValueDataTypeConverter.java

```diff
@@ -119,8 +121,8 @@ private static DataType convertToBinaryType(byte[] bytes) {
     }
 
     private static DataType convertToDecimalType(BigDecimal decimal) {
-        final int precision = decimal.precision();
         final int scale = decimal.scale();
+        final int precision = max(decimal.precision(), scale);
```

Comment:
Please add a test.

Reply:
A pull request regarding this issue has been merged (see [PR #25404](https://github.com/apache/flink/pull/25404)). However, I have some concerns about its solution. Could you please have a look?